sdks/python/apache_beam/ml/inference/huggingface_inference.py

```diff
@@ -638,6 +647,12 @@ def __init__(
     if max_batch_size is not None:
       self._batching_kwargs['max_batch_size'] = max_batch_size
     self._large_model = large_model
+    if 'device' not in self._load_pipeline_args:
+      self._load_pipeline_args['device'] = self._device
+    else:
+      _LOGGER.warning(
+          '`device` specified in `load_pipeline_args`. `device` '
+          'parameter for HuggingFacePipelineModelHandler will be ignored.')
```

Comment:
I think we should only warn if a user explicitly passes in both device as an arg and defines a device in `self._load_pipeline_args`. Otherwise this is warning at a totally valid configuration.

As a result, I think we should default to `device: str = None` above, and assume `None` means `GPU` (and doc it accordingly)

Reply:
That also allows us to warn if the user specifies `GPU` but there's no gpu available

Reply:
Done, thanks!